shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizResolver.kt

```diff
@@ -167,11 +167,20 @@ object StepQuizResolver {
         }
     }
 
-    internal fun isGptCodeGenerationWithErrorsAvailable(
+    internal fun isMobileGptCodeGenerationWithErrorsAvailable(
         step: Step,
-        isMobileGptCodeGenerationWithErrorsEnabled: Boolean
-    ): Boolean =
-        isMobileGptCodeGenerationWithErrorsEnabled &&
-            BlockName.codeRelatedBlocksNames.contains(step.block.name) &&
-            !step.isIdeRequired()
+        submissionState: StepQuizFeature.SubmissionState,
+        isMobileGptCodeGenerationWithErrorsEnabled: Boolean,
+        isMobileGptCodeGenerationWithErrorsOnboardingShown: Boolean
+    ): Boolean {
+        if (!isMobileGptCodeGenerationWithErrorsEnabled || isMobileGptCodeGenerationWithErrorsOnboardingShown) {
+            return false
+        }
+
+        if (submissionState !is StepQuizFeature.SubmissionState.Empty) {
+            return false
+        }
+
+        return BlockName.codeRelatedBlocksNames.contains(step.block.name) && !isIdeRequired(step, submissionState)
+    }
```

Comment:
This function is waned by detekt. `ReturnCount: Function isMobileGptCodeGenerationWithErrorsAvailable has 3 return statements which exceeds the limit of 2.`.
Maybe it's better to replace multiple return with `when`.